File: UI/pc-part-picker/src/model/Powersupply.ts
import { FormfactorType } from "./FormfactorType";
import { ManufacturerType } from "./ManufacturerType";
import { PcPart } from "./Pcpart";

export interface Powersupply extends PcPart{
    type: string | null;
    efficiency: string | null;
    wattage: number | null;
    modular: string | null;
    color: string | null;
    manufacturerTypeId: number | null;
    formFactorId: number | null;
    formFactor: string | null;
    manufacturerType: string | null;
}

File: API/PCPartPicker/Models/Powersupply.cs
using System;
using System.Collections.Generic;

namespace PCPartPicker.Models;

public partial class Powersupply
{
    public int Id { get; set; }

    public string? Name { get; set; }

    public decimal? Price { get; set; }

    public string? Type { get; set; }

    public string? Efficiency { get; set; }

    public decimal? Wattage { get; set; }

    public string? Modular { get; set; }

    public string? Color { get; set; }

    public int? ManufacturerTypeId { get; set; }

    public int? FormFactorId { get; set; }

    public virtual ICollection<Configuration> Configurations { get; set; } = new List<Configuration>();

    public virtual FormfactorType? FormFactor { get; set; }

    public virtual ManufacturerType? ManufacturerType { get; set; }
}
